Tests/Unit/TestHelpers.swift

```diff
@@ -0,0 +1,107 @@
+    var mockAppName: String?
+    var mockAppVersion: String?
+    var mockEndpoint: String?
+    var mockSubKey: String?
+    var mockDeviceModel: String?
+    var mockOsVersion: String?
+    var mockSdkName: String?
+    var mockSdkVersion: String?
+    var mockLanguageCode: String?
+    var mockCountryCode: String?
+    var mockNotFound: String?
+    var mockDelay: TimeInterval?
+
+    func value(for key: String) -> String? {
+        switch key {
+        case "RASApplicationIdentifier":
+            return mockAppId
+        case "RSVKeyFetchEndpoint":
+            return mockEndpoint
+        case "RASProjectSubscriptionKey":
+            return mockSubKey
+        case "CFBundleIdentifier":
+            return mockAppName
+        case "CFBundleDisplayName":
+            return mockAppName
+        case "CFBundleShortVersionString":
+            return mockAppVersion
+        default:
+            return nil
+        }
+    }
+
+    var valueNotFound: String {
+        mockNotFound ?? ""
+    }
+
+    var deviceModel: String {
+        mockDeviceModel ?? valueNotFound
+    }
+
+    var appVersion: String {
+        mockAppVersion ?? valueNotFound
+    }
+
+    var osVersion: String {
+        mockOsVersion ?? valueNotFound
+    }
+
+    var sdkName: String {
+        mockSdkName ?? valueNotFound
+    }
+
+    var sdkVersion: String {
+        mockSdkVersion ?? valueNotFound
+    }
+
+    var languageCode: String? {
+        mockLanguageCode ?? valueNotFound
+    }
+
+    var countryCode: String? {
+        mockCountryCode ?? valueNotFound
+    }
+}
+
+class FetcherMock: Fetcher {
+    var fetchConfigCalledNumTimes = 0
+    var fetchKeyCalledNumTimes = 0
+    var fetchedKey = KeyModel(data: (try? JSONSerialization.data(withJSONObject: ["id": "", "key": "", "createdAt": ""], options: []))!)
+
+    override func fetchKey(with keyId: String, completionHandler: @escaping (KeyModel?) -> Void) {
+        fetchKeyCalledNumTimes += 1
+        completionHandler(fetchedKey)
+    }
+}
+
+class VerifierMock: Verifier {
+    var verifyOK = true
+
+    override func verify(signatureBase64: String, objectData: Data, keyBase64: String) -> Bool {
+        return verifyOK
+    }
+}
+
+class APIClientMock: APIClient {
+    var data: Data?
+    var headers: [String: String]?
+    var error: Error?
+    var request: URLRequest?
+
+    override func send<T>(request: URLRequest, parser: T.Type, completionHandler: @escaping (Result<Response, Error>) -> Void) where T: Parsable {
+        self.request = request
+
+        guard let data = data, let url = request.url else {
+            return completionHandler(.failure(error ?? NSError(domain: "Test", code: 0, userInfo: nil)))
+        }
+        if let httpResponse = HTTPURLResponse(url: url, statusCode: 200, httpVersion: "1.1", headerFields: headers),
+            let object = parser.init(data: data) {
+            return completionHandler(.success(Response(object, httpResponse)))
+        }
+    }
+}
```

Comment:
As you did for `class BundleMock: EnvironmentSettable`, I think that you can create `FetcherProtocol`, `VerifierProtocol` and `APIClientProtocol` (you can find better namings) so you won't need inheritance for your mocks and it will allow you to inject dependencies with protocols.

Reply:
I agree that would be a better approach but it doesn't need to be in this PR.
Please make a ticket (or issue) for those improvements @imaginaris 

Also - Remote Config was my first Swift project 🙇 😄

Reply:
Can you create a ticket if you agree? @imaginaris

Reply:
@malayli No I won't 😄 I'll put those changes in this PR. It's not a big change